packages/data/src/api/tokenProvenance/tokenProvenance.ts

```diff
@@ -0,0 +1,44 @@
+import { fetchGraphQl } from '../../graphql/fetch';
+import { Pagination, ParsedDataReturn } from '../../types';
+import { parseData, validContractAddress, validTokenId } from '../../utils';
+import { errorContractAddress, errorToken } from './tokenProvenance.errors';
+import { getTokenProvenance } from './tokenProvenance.query';
+import { TokenProvenanceData } from './tokenProvenance.types';
+
+export const tokenProvenance = async (
+  tokenId: string | number,
+  contractAddress: string,
+  pagination?: Pagination,
+): Promise<ParsedDataReturn<TokenProvenanceData>> => {
+
+  const isValidTokenId =  validTokenId(tokenId);
+  const isValidContractAddress = validContractAddress(contractAddress);
+
+  if (!(isValidTokenId && isValidContractAddress)) {
+    if (!isValidContractAddress) {
+      console.error(errorContractAddress.message);
+
+      return { error: errorContractAddress.message };
+    }
+
+    if (!isValidTokenId) {
+      console.error(errorToken.message);
+
+      return { error: errorToken.message };
+    }
+  }
+
+  const { data, error } = await fetchGraphQl<TokenProvenanceData>({
+    query: getTokenProvenance,
+    variables: {
+      tokenId,
+      contractAddress,
+      limit: pagination?.limit ?? 12,
```

Comment:
Wonder if we should make the 12 a const like `DEFAULT_RECORDS_PER_PAGE` and add this to other data query methods